### PR TITLE
[AVAX] Merge Cortina 1 (1.10.1) 

### PR DESCRIPTION
**Pull request by evlekht**
## Why this should be merged
This PR contains changes from ava-labs/avalanchego between cortina-0 and cortina-1.
### Conflicts
All gh workflows and .md are remain unchanged in favor of our version.
- .github/packer/roles/public-ami/defaults/main.yml
- .github/packer/roles/public-ami/tasks/main.yml
- .github/workflows/amichange.json
- .github/workflows/build-public-ami.yml
- .github/workflows/test.e2e.yml
- .github/workflows/update-ami.py
- .golangci.yml
- RELEASES.md
- genesis/beacons.go
- genesis/genesis_test.go
- go.mod
- go.sum
- scripts/ansible/roles/public-ami/templates/C/config.json.j2
- scripts/ansible/ubuntu-jammy-x86_64-public-ami.json
- scripts/constants.sh
- scripts/tests.e2e.sh
- staking/tls.go
- tests/e2e/x/transfer/virtuous.go
- tests/e2e/x/whitelist-vtx/suites.go
- version/constants.go
- vms/platformvm/blocks/builder/builder_test.go
- vms/proposervm/block/block.go
### Post-merge fixes
See https://github.com/chain4travel/caminogo/pull/295#pullrequestreview-1883064673
## How this was tested
With already existing unit-tests and e2e tests.

### CODE REVIEW

File: .github/workflows/test.e2e.yml [hunk 1]
```diff
@@ -27,7 +27,4 @@ jobs:
         run: ./scripts/build.sh
       - name: Run e2e tests
         shell: bash
-        run: scripts/tests.e2e.sh ./build/camino-node ./tools/camino-network-runner/bin/camino-network-runner
-      - name: Run e2e tests for whitelist vtx
-        shell: bash
-        run: ENABLE_WHITELIST_VTX_TESTS=true ./scripts/tests.e2e.sh ./build/camino-node ./tools/camino-network-runner/bin/camino-network-runner
+        run: scripts/tests.e2e.sh ./build/camino-node ./tools/camino-network-runner/bin/camino-network-runner
```
Comment (evlekht) [inline .github/workflows/test.e2e.yml:30]: should be in merge-commit, but was somehow missed

File: scripts/tests.e2e.sh [hunk 2]
```diff
@@ -28,17 +27,18 @@ if [[ -z "${CAMINO_NETWORK_RUNNER_PATH}" ]]; then
...
 
+# Set the CGO flags to use the portable version of BLST
+#
+# We use "export" here instead of just setting a bash variable because we need
+# to pass this flag to all child processes spawned by the shell.
+export CGO_CFLAGS="-O -D__BLST_PORTABLE__"
+
+# While CGO_ENABLED doesn't need to be explicitly set, it produces a much more
+# clear error due to the default value change in go1.20.
+export CGO_ENABLED=1
```
Comment (evlekht) [inline scripts/tests.e2e.sh:40]: It is present in avax cortina, but we don't have it for some reason. Maybe it was removed way back, when blst thingies was just introduced and we thought that we can avoid using them cause of node pk feature. Anyway, having them here won't hurt and will reduce diff with avax.

File: vms/platformvm/api/camino.go [hunk 3]
```diff
@@ -320,10 +320,10 @@ func makeValidator(
 	}
 
 	if weight == 0 {
-		return nil, errValidatorAddsNoValue
+		return nil, errValidatorHasNoWeight
```
Comment (evlekht) [inline vms/platformvm/api/camino.go:323]: err was renamed in avax cortina 1

File: vms/platformvm/api/camino.go [hunk 4]
```diff
@@ -320,10 +320,10 @@ func makeValidator(
...
 	}
 	if vdr.EndTime <= vdr.StartTime {
-		return nil, errValidatorAddsNoValue
+		return nil, errValidatorAlreadyExited
```
Comment (evlekht) [inline vms/platformvm/api/camino.go:326]: err was renamed in avax cortina 1

File: vms/platformvm/api/camino_test.go [hunk 5]
```diff
@@ -23,10 +23,9 @@ import (
 )
 
 func TestBuildCaminoGenesis(t *testing.T) {
-	hrp := constants.NetworkIDToHRP[testNetworkID]
 	nodeID := ids.NodeID{1}
 	addr := ids.ShortID(nodeID)
-	addrStr, err := address.FormatBech32(hrp, addr.Bytes())
+	addrStr, err := address.FormatBech32(constants.UnitTestHRP, addr.Bytes())
```
Comment (evlekht) [inline vms/platformvm/api/camino_test.go:28]: similar to avax changes, `testNetworkID` got removed

File: vms/platformvm/txs/executor/camino_tx_executor_test.go [hunk 6]
```diff
@@ -1541,7 +1534,7 @@ func TestCaminoRewardValidatorTx(t *testing.T) {
 		generateUTXOsAfterReward: func(txID ids.ID) []*avax.UTXO {
 			return []*avax.UTXO{
 				generateTestUTXO(txID, env.ctx.AVAXAssetID, defaultCaminoValidatorWeight, stakeOwners, ids.Empty, ids.Empty),
-				generateTestUTXOWithIndex(unlockedUTXOTxID, 2, env.ctx.AVAXAssetID, defaultCaminoBalance, stakeOwners, ids.Empty, ids.Empty, true),
+				generateTestUTXOWithIndex(unlockedUTXOTxID, 1, env.ctx.AVAXAssetID, defaultCaminoBalance, stakeOwners, ids.Empty, ids.Empty, true),
```
Comment (evlekht) [inline vms/platformvm/txs/executor/camino_tx_executor_test.go:1537]: Avalanche cortina 1 changes `defaultValidateEndTime` var that is used to build test genesis, this changes validators ordering, so test removes different validator than before. And this new validator has different utxo that we expect there. Everything is meticulously investigated, compared and tested on both new and old versions.

File: vms/platformvm/utxo/camino_locked.go [hunk 7]
```diff
@@ -38,7 +38,7 @@ var (
 	errInputsUTXOsMismatch       = errors.New("number of inputs is different from number of utxos")
 	errBadCredentials            = errors.New("bad credentials")
 	errNotBurnedEnough           = errors.New("burned less tokens, than needed to")
-	errAssetIDMismatch           = errors.New("utxo/input/output assetID is different from expected asset id")
+	errUnexpectedAssetID         = errors.New("utxo/input/output assetID is different from expected asset id")
```
Comment (evlekht) [inline vms/platformvm/utxo/camino_locked.go:41]: Avalanche cortina 1 introduces new error var with exactly the same name, but with different description and meaning. This var is renamed in order to keep them segregated.

**Review by evlekht — COMMENTED** [review]
All `require.Error`, `require.ErrorContains` replacement with `require.ErrorIs` (and subsequently new error var definitions) are enforced by new linter, which is good.

File: vms/platformvm/api/camino_test.go (hunk 5 above)
Comment (knikos) [reply vms/platformvm/api/camino_test.go:28]: i see avax hasn't carried out this change consistently across all places. but it doesn't harm to update it on our end

**Review by knikos — DISMISSED** [review]

**Review by knikos — APPROVED** [review]

**Pull request merged**